Recover from broken pack file

This is a followup issue for #816: Add code to allow recovering from a broken/modified pack file.

Appreciate the tips here. I have a corrupt repo, which passes `restic check` but failes `restic check --read-data`:
```
Pack ID does not match, want a2fd8ddf0f5cc12328ad8b84d439f43ed59d33ab63bb7fb5411a404d20d51030, got 41442f67ab5ebedf513d3813222fdff6db0714a4b7499929fe8d7dc88b6a9223
Pack ID does not match, want a903774dc0c438d65f9bcf3d6523e5c50b2162b4d6fc815de524a7f209d32171, got 5250debabe1ce648075a2c046995011b7075e7bc5f1fb7d4c27b9ef6acf0a237
```
So I'm going route 1. Unfortunately I get a crash when running `restic debug examine --reupload-blobs --try-repair 41442f67ab5e6db0714a4b7499929fe8d7dc88b6a9223 5250debabe1cb7075e7bc5f1fb7d4c27b9ef6acf0a237`:

```
examine a2fd8ddf0f5cc12328ad8b84d439f43ed59d33ab63bb7fb5411a404d20d51030
  file size is 5454034
Load(<data/a2fd8ddf0f>, 0, 0) returned error, retrying after 552.330144ms: loadAll(<data/a2fd8ddf0f>): invalid data returned
Load(<data/a2fd8ddf0f>, 0, 0) operation successful after 1 retries
  wanted hash a2fd8ddf0f5cc12328ad8b84d439f43ed59d33ab63bb7fb5411a404d20d51030, got 41442f67ab5ebedf513d3813222fdff6db0714a4b7499929fe8d7dc88b6a9223
  ========================================
  looking for info in the indexes
      data blob 9ecbbdcc35046151b087429a9df9a025700756964b5a4f0ae55d1fdf6d32c5b0, offset 0     , raw length 624551
      data blob fb3543722c582da96639fa14f4b32b848408d9c05c658884354248ea3bebede7, offset 624551, raw length 4829373
      file sizes match
      loading blob 9ecbbdcc35046151b087429a9df9a025700756964b5a4f0ae55d1fdf6d32c5b0 at 0 (length 624551)
         successfully decrypted blob (length 624519), hash is 9ecbbdcc35046151b087429a9df9a025700756964b5a4f0ae55d1fdf6d32c5b0, ID matches
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x8c9065]

goroutine 1 [running]:
github.com/restic/restic/internal/repository.(*packerManager).SaveBlob(0x0, 0x1457c20, 0xc000814000, 0x610435ccbdcb9e01, 0xa0f99d9a4287b051, 0x4f5a4b9656077025, 0xc5326ddf1f5de50a, 0xc001d920b0, 0xc002200000, 0x91a0c, ...)
	/nfs/kona-home/users/darrin/src/restic/internal/repository/packer_manager.go:69 +0x65
github.com/restic/restic/internal/repository.(*Repository).saveAndEncrypt(0xc00080cb60, 0x1457c20, 0xc000814000, 0xa4f5a4b96560701, 0xc001d92000, 0x919ec, 0x98787, 0x51610435ccbdcb9e, 0x25a0f99d9a4287b0, 0xa4f5a4b96560770, ...)
	/nfs/kona-home/users/darrin/src/restic/internal/repository/repository.go:432 +0x3a7
github.com/restic/restic/internal/repository.(*Repository).SaveBlob(0xc00080cb60, 0x1457c20, 0xc000814000, 0x1, 0xc001c44010, 0x98787, 0x98797, 0x51610435ccbdcb9e, 0x25a0f99d9a4287b0, 0xa4f5a4b96560770, ...)
	/nfs/kona-home/users/darrin/src/restic/internal/repository/repository.go:843 +0x295
main.loadBlobs(0x1457c20, 0xc000814000, 0x146afc0, 0xc00080cb60, 0x23c15c0fdf8dfda2, 0x3ef439d4848bad28, 0xb57fbb63ab339dd5, 0x3010d5204d401a41, 0xc0062b8080, 0x2, ...)
	/nfs/kona-home/users/darrin/src/restic/cmd/restic/cmd_debug.go:393 +0x9dd
main.examinePack(0x1457c20, 0xc000814000, 0x146afc0, 0xc00080cb60, 0x23c15c0fdf8dfda2, 0x3ef439d4848bad28, 0xb57fbb63ab339dd5, 0x3010d5204d401a41, 0x0, 0x0)
	/nfs/kona-home/users/darrin/src/restic/cmd/restic/cmd_debug.go:519 +0x6b6
main.runDebugExamine(0x1457c20, 0xc000814000, 0xc0000460a2, 0x18, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/nfs/kona-home/users/darrin/src/restic/cmd/restic/cmd_debug.go:471 +0x335
main.glob..func9(0x1b597a0, 0xc000814200, 0x2, 0x4, 0x0, 0x0)
	/nfs/kona-home/users/darrin/src/restic/cmd/restic/cmd_debug.go:199 +0xb1
github.com/spf13/cobra.(*Command).execute(0x1b597a0, 0xc0008141c0, 0x4, 0x4, 0x1b597a0, 0xc0008141c0)
	/nfs/kona-home/users/darrin/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x49f
github.com/spf13/cobra.(*Command).ExecuteC(0x1b5bfa0, 0x23, 0xc0006c7eb8, 0x4)
	/nfs/kona-home/users/darrin/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/nfs/kona-home/users/darrin/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	/nfs/kona-home/users/darrin/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:911
main.main()
	/nfs/kona-home/users/darrin/src/restic/cmd/restic/main.go:98 +0x1d7
```

I was able to extract & repair blobs with `restic debug examine --extract-pack --try-repair a2fd8ddf0f5cc12328ad8b84d439f43ed59d33ab63bb7fb5411a404d20d51030 a903774dc0c438d65f9bcf3d6523e5c50b2162b4d6fc815de524a7f209d32171`. It found 2 blobs with bit-flip errors.

Is there a way to re-insert the extracted blobs?

(I'm using the master branch of restic, at commit 5f9ac2b1)

--- a/changelog/unreleased/issue-828
+++ b/changelog/unreleased/issue-828
@@ -6,5 +6,6 @@ on using the command if necessary to repair a repository. See the guide at
 https://restic.readthedocs.io/en/stable/077_troubleshooting.html for further
 instructions.
 
+https://github.com/restic/restic/issues/828
 https://github.com/restic/restic/pull/4644
 https://github.com/restic/restic/pull/4655

